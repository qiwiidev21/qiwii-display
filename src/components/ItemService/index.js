/*
 * ItemService Component
 */

import React, { useState, useEffect } from "react";
import "./styles.css";
import PropTypes from "prop-types";
import { useHistory, useLocation } from "react-router-dom";
import Logo from "../../assets/images/logo.png";

function ItemService({ data, index, category }) {
  const [icon, setIcon] = useState(
    data.id_icon !==
      ("https://app.qiwii.id/system/cms/themes/ace/img/logo_alt.png" || null)
      ? Logo
      : data.id_icon
  );

  const [setting, setSetting] = useState({});

  useEffect(() => {
    if (data) {
      handleJson(data);
    }
  }, [data]);

  const handleJson = (data) => setSetting(JSON.parse(data.setting));
  let history = useHistory();
  let location = useLocation();
  return (
    <div key={index} className="flex-row mx-1 d-flex my-3 card-item shadow-sm">
      <div className="col-sm-auto card-icon bg-secondary d-flex justify-content-center align-content-center p-2">
        <img
          src={icon}
          onError={() => setIcon(Logo)}
          className="rounded-circle icon-merchant"
          alt={data.unit_name}
        />
      </div>
      <div className="btn-group-vertical p-2 card-info">
        <button
          className="btn-custom btn-primary-outline"
          onClick={() =>
            history.push(`${location.pathname}/${data.id}/schedule`)
          }
        ></button>
        <button
          className="btn-custom btn-primary-outline"
          onClick={() =>
            history.push(`${location.pathname}/${data.id}/schedule`)
          }
        >
          <h6>{data.company_name}</h6>
          <h6 className="unit-name">{data.name}</h6>
        </button>
        <button
          className="btn-custom btn-primary-outline"
          onClick={() =>
            history.push(`${location.pathname}/${data.id}/schedule`)
          }
        >
          <p className="unit-address">
            Saat ini ada{" "}
            {setting.checkin === 1
              ? data.front_queue_active.checkin
              : data.front_queue_active.new}{" "}
            orang yang mengantri
          </p>
        </button>
      </div>
    </div>
  );
}

ItemService.propTypes = {
  data: PropTypes.object.isRequired,
  category: PropTypes.string.isRequired,
  index: PropTypes.number.isRequired,
};

ItemService.defaultProps = {
  data: {},
  category: "kesehatan",
  index: 0,
};

export default ItemService;
