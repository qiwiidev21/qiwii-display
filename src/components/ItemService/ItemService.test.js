// import { render } from "@testing-library/react";
import Enzyme, { shallow } from "enzyme";
import ItemService from "./index";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: () => ({
    pathname: "localhost:3000/adaptive/kesehatan",
  }),
}));

describe("<ItemService />", () => {
  const initialProps = {
    data: {
      front_queue_active: {
        checkin: "1",
        new: "1",
      },
    },
    index: 0,
    category: "kesehatan",
  };
  it("check object is true value", () => {
    const container = shallow(<ItemService {...initialProps} />);
    expect(container).toMatchSnapshot();
  });
});
