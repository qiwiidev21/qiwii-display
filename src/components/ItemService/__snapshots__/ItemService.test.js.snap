// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemService /> check object is true value 1`] = `ShallowWrapper {}`;
